### PR TITLE
Update to Webpack 5 (fixes crypto error on build)

### PR DESCRIPTION
**Pull request by SamClayton**
I updated a lot of very outdated dependencies. There are still more, but this should at least make it easier to use dependency updates to address build issues, such as the one that has been blocking Felix from building.
I've also included a containerized configuration as a standardized build environment.

There's a copy of the production site configuration deployed from my fork to [coriolis.onrender.com](https://coriolis.onrender.com/) that can be used to test this build.

### CODE REVIEW

**Comment by felixlinker** [comment]
Thanks for this PR! Much appreciated. Unfortunately, I cannot `npm install` though ☹️ 

I'm on node v18.14.1 and npm v9.3.1. Before installing, I ran `rm -rf node_modules`.

I get:
```
npm ERR! code ERESOLVE
npm ERR! ERESOLVE unable to resolve dependency tree
npm ERR!
npm ERR! While resolving: coriolis_shipyard@3.0.1
npm ERR! Found: webpack@5.75.0
npm ERR! node_modules/webpack
npm ERR!   dev webpack@"^5.75.0" from the root project
npm ERR!
npm ERR! Could not resolve dependency:
npm ERR! peer webpack@"^2.0.0 || ^3.0.0 || ^4.0.0" from workbox-webpack-plugin@4.3.1
npm ERR! node_modules/workbox-webpack-plugin
npm ERR!   dev workbox-webpack-plugin@"^4.3.1" from the root project
npm ERR!
npm ERR! Fix the upstream dependency conflict, or retry
npm ERR! this command with --force or --legacy-peer-deps
npm ERR! to accept an incorrect (and potentially broken) dependency resolution.
npm ERR!
npm ERR!
npm ERR! For a full report see:
npm ERR! /root/.npm/_logs/2023-02-19T18_19_18_363Z-eresolve-report.txt

npm ERR! A complete log of this run can be found in:
npm ERR!     /root/.npm/_logs/2023-02-19T18_19_18_363Z-debug-0.log
```

**Comment by SamClayton** [comment]
That's strange, because I've installed it numerous times, both in local Docker containers and Render.com. I do see the peer dependency mismatch, though. Can you tell me what version of npm you're running?

**Comment by SamClayton** [comment]
The service worker seems to be functioning as expected (including build-time notices about how many urls are precached) and I get no build error either on my local container or on render.com's build server.

**Comment by SamClayton** [comment]
After some more detailed testing, I'm realizing that engineering is broken in this version. I'm looking into that now.

I would still love to hear if it builds for you in its present state, @felixlinker

**Comment by SamClayton** [comment]
Okay, looks like replacing the Buffer polyfill that Webpack 5 removed has fixed the engineering issue. I'm now seeing identical values for a heavily-engineered build on both the current coriolis.io site and coriolis.onrender.com.

**Comment by felixlinker** [comment]
Sorry for the delay, was on holidays!

Yes, seems to build now 💪 

Thanks for resolving these issues!

File: src/app/Coriolis.jsx [hunk 1]
```diff
@@ -151,9 +151,9 @@ export default class Coriolis extends React.Component {
     console && console.error && console.error(arguments); // eslint-disable-line no-console
     if (errObj) {
       if (errObj instanceof Error) {
-        bugsnagClient.notify(errObj); // eslint-disable-line
+        // bugsnagClient.notify(errObj); // eslint-disable-line
       } else if (errObj instanceof String) {
-        bugsnagClient.notify(msg, errObj); // eslint-disable-line
+        // bugsnagClient.notify(msg, errObj); // eslint-disable-line
       }
     }
```
Comment (felixlinker) [inline src/app/Coriolis.jsx:158]: Can we just delete this branch?

File: src/app/index.js [hunk 2]
```diff
@@ -1,4 +1,4 @@
-import '@babel/polyfill';
+// import '@babel/polyfill';
```
Comment (felixlinker) [inline src/app/index.js:1]: Let's drop this one, too.

File: src/index.ejs [hunk 3]
```diff
@@ -47,11 +47,11 @@
 
 
     <!-- Bugsnag -->
-        <script src="https://d2wy8f7a9ursnm.cloudfront.net/v5.0.0/bugsnag.min.js"></script>
+        <!-- script src="https://d2wy8f7a9ursnm.cloudfront.net/v5.0.0/bugsnag.min.js"></script>
         <script>
       window.bugsnagClient = bugsnag('ba9fae819372850fb660755341fa6ef5', {appVersion: window.BUGSNAG_VERSION || undefined})
       window.Bugsnag = window.bugsnagClient
-    </script>
+    </script -->
```
Comment (felixlinker) [inline src/index.ejs:54]: Delete?

File: src/migrate.html [hunk 4]
```diff
@@ -25,11 +25,11 @@
       href="/192x192.png"
     />
     <!-- Bugsnag -->
-    <script src="https://d2wy8f7a9ursnm.cloudfront.net/v5.0.0/bugsnag.min.js"></script>
+    <!-- script src="https://d2wy8f7a9ursnm.cloudfront.net/v5.0.0/bugsnag.min.js"></script>
     <script>
       window.bugsnagClient = bugsnag('ba9fae819372850fb660755341fa6ef5', {appVersion: window.BUGSNAG_VERSION || undefined})
       window.Bugsnag = window.bugsnagClient
-    </script>
+    </script -->
```
Comment (felixlinker) [inline src/migrate.html:32]: Delete?

File: webpack.config.prod.js [hunk 5]
```diff
@@ -1,54 +1,39 @@
...
-const ExtractTextPlugin = require('extract-text-webpack-plugin');
-const { InjectManifest } = require('workbox-webpack-plugin');
 const CopyWebpackPlugin = require('copy-webpack-plugin');
-const { BugsnagSourceMapUploaderPlugin, BugsnagBuildReporterPlugin } = require('webpack-bugsnag-plugins');
-const pkgJson = require('./package');
-const buildDate = new Date();
+const MiniCssExtractPlugin = require('mini-css-extract-plugin');
+const { InjectManifest } = require('workbox-webpack-plugin');
+// TODO: Remove these if nobody is not monitoring Bugsnag for the project. Instantiation was already commented out below.
+// const { BugsnagSourceMapUploaderPlugin, BugsnagBuildReporterPlugin } = require('webpack-bugsnag-plugins');
```
Comment (felixlinker) [inline webpack.config.prod.js:9]: Don't even know what Bugsnag is. Happy to drop such dependencies.

**Review by felixlinker — COMMENTED** [review]
Looks all good to me. Would just propose to drop all code related to bugsnag instead of commenting it out. In one month, no one is going to remember what this does anyways.

**Comment by SamClayton** [comment]
Commented-out code removed. Should be good to go.

**Comment by felixlinker** [comment]
Thanks for your work!

**Pull request merged**